### DISCUSSION TITLE
Day 1

### DISCUSSION THREAD

**Discussion by AlecS19** [AlecS19/AoC-2024#1] [General]
Discussion about the day 1 problem and solutions for AoC

**Comment by AlecS19** [comment]
** SPOILER **
In python, it was easiest to convert the arrays to ndarrays, then sort, subtract, and compare.
Numpy arrays helped for part 2, which just involved logically indexing list2 with list1 with a for-loop. See code for example

**Comment by AlecS19** [comment]
Looking at python solution, and I am always impressed with people who do it with just basic python.
https://www.reddit.com/r/adventofcode/comments/1h3vp6n/comment/lztuvx8/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

```python
data = [*map(int, open('in.txt').read().split())]
A, B = sorted(data[0::2]), sorted(data[1::2])
print(sum(map(lambda a, b: abs(a-b), A, B)),
      sum(map(lambda a: a * B.count(a), A)))
```

Definitely should try using the map function more often.

**Comment by jaxmichalski** [comment]
I did it the exact same way as you, except I misread the question originally and did it using the argsort lol
